tools/verify_translations.py

```diff
@@ -0,0 +1,145 @@
+    return ret
+
+
+def check_duplicates(file_path: path, json_file):
+    try:
+        _ = json.load(json_file, object_pairs_hook=validate_data)
+    except ValueError as e:
+        dup_list = str(e).split(',')
+        logger.warning(f"[{path.basename(file_path)}]: duplicate {len(dup_list)}: {dup_list}")
+        return 1
+    return 0
+
+
+def check_empty_entries(file_path: path, json_file):
+    json_data = json.load(json_file)
+    empty_entries = [entry for entry in json_data if len(str(json_data[entry])) == 0]
+    if empty_entries:
+        logger.warning(f"[{path.basename(file_path)}]: empty entries {len(empty_entries)}: {empty_entries}")
+        return 1
+    return 0
+
+
+def get_all_keys_from_path(json_path: path):
+    dir_list = os.listdir(json_path)
+    json_keys = []
+
+    # iterate to get all possible keys and check for key duplicates
+    for file in dir_list:
+        file_path = path.join(json_path, file)
+
+        with open(file_path) as json_file:
+            json_data = json.load(json_file)
+            json_keys.append(set(json_data))
+
+    return set.union(*json_keys)
+
+
+def check_missing_entries_from_path(json_path: path):
+    ret = 0
+    dir_list = os.listdir(json_path)
+    all_keys = get_all_keys_from_path(json_path)
+
+    # iterate to find missing keys
+    for file in dir_list:
+        file_path = path.join(json_path, file)
+        with open(file_path) as json_file:
+            json_data = json.load(json_file)
+            missing_keys_in_file = all_keys.difference(set(json_data))
+            if missing_keys_in_file:
+                logger.warning(f"[{file}]: missing {len(missing_keys_in_file)}: {sorted(missing_keys_in_file)}")
+                ret |= 1
+    return ret
+
+
+def fix_jsons(json_path: path):
+    dir_list = os.listdir(json_path)
+    for file in dir_list:
+        file_path = path.join(json_path, file)
+        target_filename = file_path + ".tmp"
+
+        with open(file_path) as json_file, open(target_filename, 'w') as outfile:
+            json_data = json.load(json_file)
+            json.dump(json_data, outfile, indent=4, sort_keys=True)
+        os.remove(file_path)
+        os.rename(file_path + ".tmp", file_path)
+
+    logger.info("Translation files fixed")
+
+
+def main(args):
+    ret = 0
+    if args.fix:
+        fix_jsons(args.path)
+
+    ret |= check_missing_entries_from_path(args.path)
+    ret |= perform_on_files_from_path(args.path, check_empty_entries)
+    ret |= perform_on_files_from_path(args.path, check_duplicates)
+    return ret
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        prog='verify_translations',
+        description='Script for checking the inconsistency of lang jsons',
+        formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument('-p', '--path', help="path to the json files", required=True)
+    parser.add_argument('--fix', action='store_true', help=textwrap.dedent('''\
```

Comment:
How about adding optional argument(file path/name) to store the result of 'fix' operation?

Reply:
good idea, I'll add this one to prevent accidental overwriting the files

Reply:
added